src/plugins/intel_gna/src/transformations/remove_in_out_processing.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/remove_in_out_processing.hpp"
+
+#include "common/graph_utils.hpp"
+#include "openvino/cc/pass/itt.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/opsets/opset7.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "openvino/pass/manager.hpp"
+#include "openvino/pass/pass.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/transformation_helper.hpp"
+
+using namespace ov::opset10;
+using namespace ov::intel_gna::pass;
+
+namespace {
+
+inline bool is_preprocessing_layer_not_supported(std::shared_ptr<ov::Node>& layer) {
+    // Gather layer is not supported by GNA and has to be executed on CPU
+    if (std::dynamic_pointer_cast<ov::opset1::Gather>(layer) || std::dynamic_pointer_cast<ov::opset7::Gather>(layer) ||
+        std::dynamic_pointer_cast<ov::opset8::Gather>(layer)) {
+        return true;
+    }
+
+    // Verify that transpose layer cannot be executed on GNA
+    if (std::dynamic_pointer_cast<ov::opset1::Transpose>(layer)) {
+        const ov::Shape squeezed_shape = graph_utils::squeeze_shape(layer->get_shape());
+        const size_t min_input_dim = std::min(squeezed_shape[0], squeezed_shape[1]);
+        const size_t max_input_dim = std::max(squeezed_shape[0], squeezed_shape[1]);
+
+        // GNA transpose limits:
+        // - supports 2d transposes only
+        // - lesser dimention should be more then 8
+        // - greater dimention should be a multiple of limitations::noOfInputsDivisor
+        if (squeezed_shape.size() > 2) {
+            return true;
+        } else if (min_input_dim > 8) {
+            return true;
+        } else if (ALIGN(max_input_dim, limitations::noOfInputsDivisor) != max_input_dim) {
```

Comment:
I suggest to not tie transposition dimensions to number of inputs divisor. I don't think it's guaranteed to be related to each other.